File: frontend/public/generate_icons.py
#!/usr/bin/env python3
"""
Generate PWA icons from favicon.ico
This script creates placeholder PNG icons for the PWA manifest.
"""

from PIL import Image, ImageDraw, ImageFont
import os

# Margav Energy color scheme
PRIMARY_COLOR = (51, 51, 204)  # #3333cc
BLACK = (0, 0, 0)  # #000000
WHITE = (255, 255, 255)  # #ffffff
GREEN = (102, 204, 102)  # #66cc66

def create_icon(size, filename):
    """Create a simple icon with Margav Energy branding"""
    # Create image with transparent background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Draw background circle
    margin = size // 8
    draw.ellipse([margin, margin, size-margin, size-margin], 
                 fill=PRIMARY_COLOR, outline=BLACK, width=max(1, size//64))
    
    # Add text "ME" for Margav Energy
    try:
        # Try to use a font, fallback to default if not available
        font_size = size // 3
        try:
            font = ImageFont.truetype("arial.ttf", font_size)
        except:
            font = ImageFont.load_default()
    except:
        font = ImageFont.load_default()
    
    # Calculate text position
    text = "ME"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    x = (size - text_width) // 2
    y = (size - text_height) // 2
    
    # Draw text
    draw.text((x, y), text, fill=WHITE, font=font)
    
    # Save the image
    img.save(filename, 'PNG')
    print(f"Created {filename} ({size}x{size})")

def main():
    """Generate all required PWA icons"""
    # Ensure icons directory exists
    os.makedirs('icons', exist_ok=True)
    
    # Required icon sizes
    sizes = [72, 96, 128, 144, 152, 192, 384, 512]
    
    print("Generating PWA icons from favicon...")
    
    for size in sizes:
        filename = f'icons/icon-{size}x{size}.png'
        create_icon(size, filename)
    
    print("\n✅ All PWA icons generated successfully!")
    print("Icons created:")
    for size in sizes:
        print(f"  - icons/icon-{size}x{size}.png")
    
    print("\n📱 Your PWA is now ready with proper icons!")
    print("You can replace these with converted versions of your favicon.ico later.")

if __name__ == "__main__":
    main()

File: frontend/public/generate-icons.js
const fs = require('fs');
const path = require('path');

// Create icons directory if it doesn't exist
const iconsDir = path.join(__dirname, 'icons');
if (!fs.existsSync(iconsDir)) {
    fs.mkdirSync(iconsDir, { recursive: true });
}

// Simple SVG-based icon generator
function createSVGIcon(size) {
    return `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${size}" height="${size}" viewBox="0 0 ${size} ${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <style>
      .circle { fill: #3333cc; stroke: #000000; stroke-width: ${Math.max(1, size/64)}; }
      .text { fill: #ffffff; font-family: Arial, sans-serif; font-size: ${size/3}px; font-weight: bold; text-anchor: middle; dominant-baseline: central; }
    </style>
  </defs>
  <circle cx="${size/2}" cy="${size/2}" r="${size/2 - size/8}" class="circle"/>
  <text x="${size/2}" y="${size/2}" class="text">ME</text>
</svg>`;
}

// Required icon sizes
const sizes = [72, 96, 128, 144, 152, 192, 384, 512];

console.log('🎨 Generating PWA icons for Margav Energy...\n');

sizes.forEach(size => {
    const svgContent = createSVGIcon(size);
    const filename = path.join(iconsDir, `icon-${size}x${size}.svg`);
    
    fs.writeFileSync(filename, svgContent);
    console.log(`✅ Created icon-${size}x${size}.svg`);
});

console.log('\n📱 PWA icons generated successfully!');
console.log('\n📋 Next steps:');
console.log('1. Open each SVG file in a browser or image editor');
console.log('2. Export as PNG format');
console.log('3. Replace the .svg files with .png files');
console.log('4. Your PWA will now have proper icons!');
console.log('\n🎯 Alternative: Use an online SVG to PNG converter:');
console.log('   - https://convertio.co/svg-png/');
console.log('   - https://cloudconvert.com/svg-to-png');

